server/src/main/java/io/crate/expression/predicate/NotPredicate.java

```diff
@@ -161,8 +179,11 @@ public Void visitFunction(Function function, NullabilityContext context) {
                     return null;
                 }
             }
+            var savedContext = context.childContext();
             for (Symbol arg : function.arguments()) {
-                arg.accept(this, context);
+                var childContext = savedContext.childContext();
+                arg.accept(this, childContext);
+                context.accumulate(childContext);
```

Comment:
accumulating the context is only there to keep the original behaviour of the top-level NullabilityContext.